integrated_channels/integrated_channel/tasks.py

```diff
@@ -73,110 +104,102 @@ def transmit_learner_data(username, channel_code, channel_pk):
-        enterprise_customer_identifier=api_user.username
-    )
+    _log_batch_task_finish('transmit_learner_data', channel_code, api_user.id, integrated_channel, duration)
 
 
 @shared_task
 @set_code_owner_attribute
-def transmit_single_learner_data(username, course_run_id):
+def transmit_single_learner_data(learner_username, course_run_id):
     """
-    Task to send single learner data to each linked integrated channel.
+    Task to send learner data to each linked integrated channel.
 
     Arguments:
-        username (str): The username of the learner whose data it should send.
+        learner_username (str): The username of the learner whose data it should send.
         course_run_id (str): The course run id of the course it should send data for.
     """
-    user = User.objects.get(username=username)
+    user = User.objects.get(username=learner_username)
     enterprise_customer_uuids = get_enterprise_uuids_for_user_and_course(user, course_run_id, active=True)
 
     # Transmit the learner data to each integrated channel for each related customer.
     # Starting Export. N customer is usually 1 but multiple are supported in codebase.
     for enterprise_customer_uuid in enterprise_customer_uuids:
-        LOGGER.info('[Integrated Channel] Single learner data transmission started.'
-                    ' Course: {course_run}, Username: {username}, Customer:{enterprise_uuid}'.format(
-                        course_run=course_run_id,
-                        username=username,
-                        enterprise_uuid=enterprise_customer_uuid
-                    ))
-
         channel_utils = IntegratedChannelCommandUtils()
-        # Transmit the learner data to each integrated channelStarting Export
-        for channel in channel_utils.get_integrated_channels(
-                {'channel': None, 'enterprise_customer': enterprise_customer_uuid}
-        ):
+        enterprise_integrated_channels = channel_utils.get_integrated_channels(
+            {'channel': None, 'enterprise_customer': enterprise_customer_uuid}
+        )
+        for channel in enterprise_integrated_channels:
             integrated_channel = INTEGRATED_CHANNEL_CHOICES[channel.channel_code()].objects.get(pk=channel.pk)
-            LOGGER.info(
-                '[Integrated Channel] Processing learner for transmission. Configuration: {configuration},'
-                ' User: {user_id}, Customer: {enterprise_uuid}'.format(
-                    configuration=integrated_channel,
-                    user_id=user.id,
-                    enterprise_uuid=enterprise_customer_uuid))
+
+            LOGGER.info(generate_formatted_log(
+                integrated_channel.channel_code(),
+                enterprise_customer_uuid,
+                user.id,
+                course_run_id,
+                'transmit_single_learner_data for channels {} started.'
+                .format([c.channel_code() for c in enterprise_integrated_channels])
```

Comment:
Isn't this (and the below log line) going to be generated for each channel? So why would you use a loop here?

Ie we'll see like 4-6 log lines that consist of:
`transmit_single_learner_data for channels Canvas, Blackboard, Moodle, etc. started.` 
That seems extremely redundant and I imagine them being separated helps us verify that a particular channel did actually attempt to transmit something.

Reply:
I see the below you added for the `finish` **does** use the specific channel code, so not sure why we're not doing the same here.

Reply:
Probably just a mistake on my part, should just be a single channel.